src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -892,6 +892,24 @@ private String createProgramGroupID(final SAMFileHeader header) {
 
         final FeatureInput<? extends Feature> featureInput = new FeatureInput<>(filePath, name);
 
+        //Add datasource to the feature manager too so that it can be queried. Setting lookahead to 0 to avoid caching.
+        //Note: we are disabling lookahead here because of windowed queries that need to "look behind" as well.
+        addFeatureInputsAfterInitialization(featureInput, featureType, featureQueryLookahead);
+
+        return featureInput;
+    }
+
+    /**
+     * A method to allow a user to inject data sources after initialization that were not specified as command-line
+     * arguments.
+     *
+     * @param featureInput
+     * @param featureType class of feature in the feature input
+     * @param featureQueryLookahead look ahead this many bases during queries that produce cache misses
+     * @return The {@link FeatureInput} used as the key for this data source.
+     */
+    protected void addFeatureInputsAfterInitialization(final FeatureInput<? extends Feature> featureInput,
+                                                                                  final Class<? extends Feature> featureType, final int featureQueryLookahead) {
         //Add datasource to the feature manager too so that it can be queried. Setting lookahead to 0 to avoid caching.
         //Note: we are disabling lookahead here because of windowed queries that need to "look behind" as well.
```

Comment:
Comment is wrong here too -- lookahead caching is not disabled here.

Reply:
Fixed!